include/field.hxx

```diff
@@ -121,11 +140,65 @@ class Field {
    */
   virtual int getNz() const;
 
+  friend void swap(Field& first, Field& second) noexcept {
+    using std::swap;
+    swap(first.name, second.name);
+    swap(first.fieldmesh, second.fieldmesh);
+    swap(first.fieldCoordinates, second.fieldCoordinates);
+    swap(first.location, second.location);
+    swap(first.directions, second.directions);
+  }
+
+  friend bool fieldsCompatible(const Field& field1, const Field& field2) {
+    return
+        // The following is a possible alternative to
+        // checking the two meshes and location are the same
+        // It is slightly stricter if we decide that coordinates
+        // could differ in more than just location.
+        field1.getCoordinates() == field2.getCoordinates() &&
+        // In the unit tests fieldCoordinates get set to nullptr, so we still
+        // need to check fieldmesh and location, at least for now
+        field1.getMesh() == field2.getMesh() &&
+        field1.getLocation() == field2.getLocation() &&
+        // Compatible directions
+        compatibleDirections(field1.directions, field2.directions);
+  }
 protected:
   Mesh* fieldmesh{nullptr};
   mutable std::shared_ptr<Coordinates> fieldCoordinates{nullptr};
+
+  /// Location of the variable in the cell
+  CELL_LOC location{CELL_CENTRE};
+
+  /// Copy the members from another Field
+  void copyFieldMembers(const Field& f) {
+    name = f.name;
+    fieldmesh = f.fieldmesh;
+    fieldCoordinates = f.fieldCoordinates;
+    location = f.location;
+    directions = f.directions;
+  }
+
+  /// Setters for *DirectionType
+  void setDirectionY(YDirectionType y_type) {
+    directions.y = y_type;
+  }
+  void setDirectionZ(ZDirectionType z_type) {
+    directions.z = z_type;
+  }
+
+private:
+  DirectionTypes directions{YDirectionType::Standard, ZDirectionType::Standard};
 };
 
+/// Return an empty shell field of some type derived from Field, with metadata
+/// copied but empty data array
```

Comment:
Please could you add something explicit here about the field being allocated but uninitialised?